### PR TITLE
default interpreter sub-tests to run in parallel

### PR DESCRIPTION
**Pull request by a7i**
Signed-off-by: Amir Alavi <amiralavi7@gmail.com>

**What type of PR is this?**
/king cleanup

**What this PR does / why we need it**:
Continuation of #2995
Make the rest of the defaultinterpreter tests to run in parallel

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ikaven1024** [comment]
https://github.com/karmada-io/karmada/actions/runs/3788983014/jobs/6442379846#step:4:1072 this log show the mistake in your test.
When we range iterates a go slice, the second parameter is shared. You can test it with:
```go
func Test1(t *testing.T) {
	tests := []struct{ name string }{
		{name: "test1"},
		{name: "test2"},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			t.Parallel()
			t.Log(tt.name)
		})
	}
}
```

The two gorputines has the same value of `tt`. You can see the outputs:
```
...
=== CONT  Test1/test1
    revisereplica_test.go:169: test2
=== CONT  Test1/test2
    revisereplica_test.go:169: test2
...
```

---

You shall use the first parameter in rang:
```go
...
	for i := range tests {
		tt := tests[i]
		t.Run(tt.name, func(t *testing.T) {
			t.Parallel()
			t.Log(tt.name)
		})
	}
```
And output：
```
=== CONT  Test2/test1
    revisereplica_test.go:184: test1
=== CONT  Test2/test2
    revisereplica_test.go:184: test2
```

File: pkg/resourceinterpreter/defaultinterpreter/dependencies_test.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ func TestGetSecretNames(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (ikaven1024) [inline pkg/resourceinterpreter/defaultinterpreter/dependencies_test.go:49]: Only the last sub test will run~

**Comment by a7i** [comment]
Good catch @ikaven1024 should be fixed now.

**Review by RainbowMango — APPROVED** [review]
/approve
It should be ok.

**Review by RainbowMango — APPROVED** [review]
/lgtm

**Pull request merged**